sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TimeWindow.scala

```diff
@@ -103,9 +103,9 @@ object TimeWindow {
    */
   private def getIntervalInMicroSeconds(interval: String): Long = {
     val cal = CalendarInterval.fromCaseInsensitiveString(interval)
-    if (cal.months > 0) {
+    if (cal.months > 0 || cal.days > 0) {
       throw new IllegalArgumentException(
-        s"Intervals greater than a month is not supported ($interval).")
+        s"Intervals greater than a day is not supported ($interval).")
```

Comment:
`TimeWindow` use `Long` to represent `window, slide, startTime`, if we change `Long` to `CalendarInterval`, we can reduce this constraint